### DISCUSSION TITLE
Fresh installation with Nginx Proxy Manager

### DISCUSSION THREAD

**Discussion by Kevin-Nexio** [openrport/openrport#12] [Q&A]
Hello,

First of all I would like to thank you for the incredible work you do and for continuing to make rPort open source!

I'm trying to make a new installation (it's my first rPort installation...) with an openRport on an Ubuntu 22.04 behind a Nginx Proxy Manager.
The aim is to be able to manage IOT devices (SSH and HTTPS) from rPort.
I have one constraint: since I don't know the network configuration of my clients or where IOT devices will be used, I need a solution that goes everywhere, so I'd like everything to go through port 443 on the client side.

I also want everything to go through nginx so I don't have to do any routing from my firewall.

For the moment, I've done the following:

```terminal 
sudo bash rportd-installer.sh \
 --totp \
 --client-port 443 \
 --api-port 5000 \
 --fqdn rport.nexdesk.ch \
 --port-range 20000-20050 \
 --client-url https://client.nexdesk.ch
```

Installation went smoothly. 

Then I added 2 Proxy Hosts on my Nginx:

**1.**
<img width="498" alt="image" src="https://github.com/openrport/openrport/assets/97556752/17600319-fcff-49bb-8452-4a3f89c7cf3c">
<img width="498" alt="image" src="https://github.com/openrport/openrport/assets/97556752/c4b8885e-914e-42a1-afef-1a151e9975c8">
<img width="498" alt="image" src="https://github.com/openrport/openrport/assets/97556752/bd56ac6c-398f-4da5-a421-1f2d23e0b66f">

**2.**
<img width="498" alt="image" src="https://github.com/openrport/openrport/assets/97556752/d2675e78-d836-4998-9e24-c6da4142a5ca">
<img width="498" alt="image" src="https://github.com/openrport/openrport/assets/97556752/3aa75514-6e88-4f95-a748-611dfb1e0eed">
<img width="498" alt="image" src="https://github.com/openrport/openrport/assets/97556752/8b495ad3-d864-455b-b84a-a66e4429f0c2">

I'm able to connect to the Web interface and everything seems to be working fine.
However, when I want to install a client (on a DietPi ARM) I get an error:

```terminal
root@DietPi:~# sudo sh rport-installer.sh -x -s -r -b -i
Installing rport client
Detected Linux Distribution: Armbian 23.8.3 bookworm
Success: https://client.nexdesk.ch is reachable. All good.
[*] System is already using the rport deb repo.
Hit:1 http://repo.rport.io/deb bookworm InRelease
Hit:2 https://deb.debian.org/debian bookworm InRelease                                            
Hit:3 https://deb.debian.org/debian bookworm-updates InRelease                                    
Hit:4 https://deb.debian.org/debian-security bookworm-security InRelease
Hit:5 https://deb.debian.org/debian bookworm-backports InRelease
Hit:6 https://mirror.funkfreundelandshut.de/armbian/apt bookworm InRelease
Reading package lists... Done
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
rport is already the newest version (1.0.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Success: RPort will run as user rport
Success: User rport already exist.
Success: /etc/rport/rport.conf created.
Preparing /etc/rport/rport.conf
Using /etc/machine-id as rport client id
  net_lan = ['eth0' , '1000' ]
Interpreter 'bash' found in '/usr/bin/bash'
Interpreter 'bash' already registered.
Interpreter 'perl' found in '/usr/bin/perl'
Interpreter 'perl' already registered.
Systemd service already present.
[*] You already have a /etc/sudoers.d/rport-update-status. Not changing.
[*] You already have a /etc/sudoers.d/rport-all-cmd. Not changing.
[*] Tacoscript already installed. Checking for updates ...
[*] Nothing to do. Tacoscript is on the latest version 0.3.0.
2023/10/07 10:43:55 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 10 of infinite)
2023/10/07 10:33:55 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 9 of infinite)
2023/10/07 10:31:17 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 8 of infinite)
2023/10/07 10:26:22 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 7 of infinite)
2023/10/07 10:26:00 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 6 of infinite)
2023/10/07 10:24:07 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 5 of infinite)
2023/10/07 10:23:48 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 4 of infinite)
2023/10/07 10:23:15 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 3 of infinite)
2023/10/07 10:23:06 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 2 of infinite)
2023/10/07 10:22:59 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 1 of infinite)
2023/10/07 10:22:54 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies

#
# -------------!!   ERROR  !!-------------
#
# Installation of rport finished with errors.
#

Try the following to investigate:
1) systemctl rport status

2) tail /var/log/rport/rport.log

3) Ask for help on https://kb.rport.io/need-help/request-support
```
```consol
root@DietPi:~# tail /var/log/rport/rport.log
2023/10/07 10:31:17 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 8 of infinite)
2023/10/07 10:31:17 info: client: Retrying in 2m37.668666555s...
2023/10/07 10:33:54 info: client: Trying to connect to ws://rport.nexdesk.ch:80 ...
2023/10/07 10:33:54 info: client: Will wait up to 300.00 seconds for the server to respond
2023/10/07 10:33:55 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 9 of infinite)
2023/10/07 10:33:55 info: client: Retrying in 10m0s...
2023/10/07 10:43:55 info: client: Trying to connect to ws://rport.nexdesk.ch:80 ...
2023/10/07 10:43:55 info: client: Will wait up to 300.00 seconds for the server to respond
2023/10/07 10:43:55 error: client: connection error: websocket: bad handshake - Server maybe busy. Also check your client credentials AND check for tranparent proxies (attempt: 10 of infinite)
2023/10/07 10:43:55 info: client: Retrying in 10m0s...
```
Can you help me?
If I've understood correctly, I also need to open the 20000-20050 on my firewall, can I also direct them to my nginx and from there redirect them to my ubuntu machine with rPort? If so, do I need to create Streams?

I hope I'm clear enough and that you understand my idea and that someone can help me...

Thanks in advance

**Comment by alhdo** [comment]
Hello @Kevin-Nexio thanks for reporting this. We will try to help you on this issue. We ll try to reproduce it on our end and come back with a solution.

**Reply by Kevin-Nexio** [reply]
Ok thanks you very much !

**Comment by buzzzo** [comment]
The proxy part (so the http/https) is used only to proxy the web ui. 
The range of ports is used directly by the rportd service and on them it will be proxy your remote rport client.

For example, assuming that rport.foo.bar is your rport server,based on how you configure the services you could have:
1) an rdp remote service of client1 proxied on rport.foo.bar: 20000
2) a http remote service  of client2 proxied on rport.foo.bar port 20001
and so on.

The port range has never proxied by the reverse proxy of choice

**Comment by Kevin-Nexio** [comment]
Hello,

Thank you, so if I understand correctly ports 20'000 to 20'050 can't go through the Nginx reverse proxy?
For the rest, does this configuration seem ok?
<img width="1693" alt="image" src="https://github.com/openrport/openrport/assets/97556752/42f38875-0eeb-4a21-85a8-8900130606a1">

**Reply by buzzzo** [reply]
Yes exactly.
Remember that when you open the port range on your firewall you basically open services forwarded on it by rport to the world .

Place ACL on the service settings and close time

**Comment by sviktor75** [comment]
> Hello,
> 
> First of all I would like to thank you for the incredible work you do and for continuing to make rPort open source!
> 
> I'm trying to make a new installation (it's my first rPort installation...) with an openRport on an Ubuntu 22.04 behind a Nginx Proxy Manager. The aim is to be able to manage IOT devices (SSH and HTTPS) from rPort. I have one constraint: since I don't know the network configuration of my clients or where IOT devices will be used, I need a solution that goes everywhere, so I'd like everything to go through port 443 on the client side.
> 
> I also want everything to go through nginx so I don't have to do any routing from my firewall.
> 
> For the moment, I've done the following:
> 
> ```
> sudo bash rportd-installer.sh \
>  --totp \
>  --client-port 443 \
>  --api-port 5000 \
>  --fqdn rport.nexdesk.ch \
>  --port-range 20000-20050 \
>  --client-url https://client.nexdesk.ch
> ```
> 
> Installation went smoothly.
> 

You can not use port 443 as client-port.
Use instead 80, because corporate networks usually do not block outgoing traffic on port 80. Plain HTTP is used, no certificates are needed. The encryption happens on application level and client and server handle it autonomously based on the SSH protocol over HTTP.
You're trying to connect to ws://clients.example.com:443, but port 443 is typically used for HTTPS/WSS (secure WebSocket connections). The protocol mismatch is likely causing the bad handshake.
client try to connect to ws:// - nginx handle at 443 only wss:// - server also accepts only ws:// connections
Use any other port. The best is as I wrote up there port 80.
<img width="610" height="664" alt="2025  09  29" src="https://github.com/user-attachments/assets/9f8ed765-5404-4e57-a2af-54819c2fe290" />
<img width="614" height="471" alt="2025  09  292" src="https://github.com/user-attachments/assets/84300fa3-1375-40bc-8504-38355609bfa1" />
